src/world/generator/object/NetherTree.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\world\generator\object;
+
+use pocketmine\block\Block;
+use pocketmine\block\NetherFungus;
+use pocketmine\block\NetherVines;
+use pocketmine\block\VanillaBlocks;
+use pocketmine\utils\Random;
+use pocketmine\world\BlockTransaction;
+use pocketmine\world\ChunkManager;
+use function abs;
+use function min;
+
+class NetherTree extends Tree{
+
+	private bool $huge;
+
+	public function __construct(Block $stemBlock, Block $hatBlock, Random $random, private readonly bool $hasVines, private readonly bool $planted = false){
+		$i = $random->nextBoundedInt(9) + 4;
+		if($random->nextBoundedInt(12) === 0){
+			$i *= 2;
+		}
+		$this->huge = !$this->planted && $random->nextFloat() < 0.06;
+		parent::__construct($stemBlock, $hatBlock, $i);
+	}
+
+	public function canPlaceObject(ChunkManager $world, int $x, int $y, int $z, Random $random) : bool{
+		if($this->planted){
+			return true;
+		}
+		return parent::canPlaceObject($world, $x, $y, $z, $random);
+	}
+
+	public function getBlockTransaction(ChunkManager $world, int $x, int $y, int $z, Random $random) : ?BlockTransaction{
+		if(!$this->canPlaceObject($world, $x, $y, $z, $random)){
+			return null;
+		}
+
+		$transaction = new BlockTransaction($world);
+		$this->placeStem($x, $y, $z, $random, $this->generateTrunkHeight($random), $transaction);
+		$this->placeHat($x, $y, $z, $random, $transaction);
+
+		return $transaction;
+	}
+
+	protected function placeStem(int $x, int $y, int $z, Random $random, int $trunkHeight, BlockTransaction $transaction) : void{
+		$i = $this->huge ? 1 : 0;
+
+		for($j = -$i; $j <= $i; ++$j){
+			for($k = -$i; $k <= $i; ++$k){
+				$isCorner = $this->huge && abs($j) === $i && abs($k) === $i;
+
+				for($l = 0; $l < $trunkHeight; ++$l){
+					$blockX = $x + $j;
+					$blockY = $y + $l;
+					$blockZ = $z + $k;
+
+					if($this->canOverride($transaction->fetchBlockAt($blockX, $blockY, $blockZ))){
+						if($isCorner){
+							if($random->nextFloat() < 0.1){
+								$transaction->addBlockAt($blockX, $blockY, $blockZ, $this->trunkBlock);
+							}
+						}else{
+							$transaction->addBlockAt($blockX, $blockY, $blockZ, $this->trunkBlock);
+						}
```

Comment:
This could be simplified to `if(!$isCorner || $random->nextFloat() < 0.1)`, not needing an `else` branch (and could also be merged into the outer condition)